### PR TITLE
SFR-1232: Adding integration and e2e tests for FirstBook login

### PR DESCRIPTION
**Pull request by samanthaandrews**
I'm so sorry, I accidentally closed the previous pull request D: tried to squash my commits into one and then didn't commit them correctly so the branch closed itself lol I've never had that happen before!

Wrote both integration and e2e tests for:
- Ensuring the various login pages display correctly when navigating around open eBooks
- Manually logging into FirstBook, i.e. typing in the username and password
- Ensuring error messages appear when entering only password, only username, or incorrect credentials
- Logging out using the "logout" button

### CODE REVIEW

File: cypress/tests/open-ebooks/integration/login.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  it("Displays error when entering incorrect credentials", () => {
+    cy.visit(FIRSTBOOK_LOGIN_PATH);
+    cy.findByRole("textbox", { name: "Access Code input" }).type(
+      Cypress.env("OPENEBOOKS_ALL_ACCESS_USER_USERNAME"),
+      { log: false }
+    );
+    cy.findByLabelText("PIN input").type("1234", { log: false });
+
+    // TODO: Figure out why this mock is causing error message to not display
+    // I think it has to do with the swr in UserContext.tsx
```
Comment (samanthaandrews) [inline cypress/tests/open-ebooks/integration/login.ts:134]: Let's pair on this to make sure I understand the flow of this request. I'm unable to mock the correct request or response in this integration test, which makes me think there's something I'm missing.

File: cypress/tests/open-ebooks/integration/login.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    ).should("exist");
+  });
+
+  it("Can logout with FirstBook", () => {
+    cy.loginByApi("ALL_ACCESS_USER");
+    cy.visit(APP_PATH);
+    cy.wait(5000);
+
+    // TODO: Figure out why this test fails when I try to mock the response
+    // cy.intercept("POST", "/").as("logout");
```
Comment (samanthaandrews) [inline cypress/tests/open-ebooks/integration/login.ts:151]: Let's also pair to get this one working. I'm not quite understanding the flow of the requests that are being made which is making it hard to get this test to pass with mock intercepts & data. As you can see in this test failure screenshot, a few POST requests seem to be happening, which makes complete sense, but I haven't been able to correctly mock the behavior in the tests.

FirstBook login -- Can logout with FirstBook (failed) copy

File: cypress/tests/open-ebooks/integration/login.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      .findByRole("heading", { name: "Open eBooks (QA Server) Home" })
+      .should("exist");
+    cy.findByRole("heading", { name: "High School collection" }).should(
+      "exist"
+    );
+  });
+
+  it("Redirects to collection page after login", () => {
+    // FIXME: Redirect is not happening when first navigating to a collection and then logging in
+    // cy.visit(ALL_ACCESS_AUTHOR_RECOMMENDATIONS_PATH_HUNTER_C_C);
```
Comment (samanthaandrews) [inline cypress/tests/open-ebooks/integration/login.ts:95]: Jira issue to track this: https://jira.nypl.org/browse/OE-197

File: cypress/tests/open-ebooks/server-e2e/login.ts [hunk 4]
```diff
@@ -1,61 +1,153 @@
...
+    cy.get("main")
+      .findByRole("heading", { name: "Open eBooks (QA Server) Home" })
+      .should("exist");
+    cy.findByRole("heading", { name: "High School collection" }).should(
+      "exist"
+    );
+  });
+
+  it("Redirects to collection page after login", () => {
+    // FIXME: Redirect is not happening when first navigating to a collection and then logging in
```
Comment (samanthaandrews) [inline cypress/tests/open-ebooks/server-e2e/login.ts:81]: Jira issue to track this: https://jira.nypl.org/browse/OE-197

**Comment by samanthaandrews** [comment]
Ok, @kristojorg and @Toxiapo, this is ready for review. I _think_ I figured out the check to ensure the Cypress Tests check is reporting correctly on each commit (I believe the problem was a missing comma in my JSON post request 😩 ).

File: .github/workflows/oe-integration-cypress.yml [hunk 5]
```diff
@@ -99,9 +99,9 @@ jobs:
           --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
           --header 'Accept: application/vnd.github.v3+json' \
           --data '{
-            "context": "Cypress Tests",
+            "context": "Cypress Tests - Open eBooks",
             "state": "success",
-            "description": "Cypress tests passed"
+            "description": "Cypress tests passed",
```
Comment (samanthaandrews) [inline .github/workflows/oe-integration-cypress.yml:104]: this missing comma.................... was the problem the whole time

Comment (kristojorg) [reply .github/workflows/oe-integration-cypress.yml:104]: Github workflows is truly a nightmare of debugging

**Review by kristojorg — APPROVED** [review]
Looks good to me!

File: cypress/fixtures/open-ebooks/all-access/all-collections.html [hunk 6]
```diff
@@ -1,1990 +1,2216 @@
...
-    <link href="http://openebooks.net/resources.html" title="Resources" type="text/html" rel="related" role="navigation"/>
-    <link href="https://openebooks.zendesk.com/" title="Help Center" type="text/html" rel="related" role="navigation"/>
-    <link href="mailto:leonardrichardson@nypl.org" rel="help"/>
-  </feed>
-  
\ No newline at end of file
+  <id>https://qa-circulation.openebooks.us/USOEI/groups/?entrypoint=Book</id>
+  <title>Open eBooks (QA Server)</title>
+  <updated>2021-08-26T18:50:05+00:00</updated>
+  <link href="https://qa-circulation.openebooks.us/USOEI/groups/?entrypoint=Book" rel="self"/>
```
[the remark this answers is not in the document]
Comment (samanthaandrews) [reply cypress/fixtures/open-ebooks/all-access/all-collections.html:5]: Technically it's an xml file, but Cypress can't currently accept that type of file as a fixture.

**Pull request merged**